halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1508,6 +1623,31 @@ impl<F: Field, C: Into<Constraint<F>>, Iter: IntoIterator<Item = C>> IntoIterato
     }
 }
 
+/// GateV2Backend
+#[derive(Clone, Debug)]
+pub struct GateV2Backend<F: Field> {
+    name: String,
+    constraint_names: Vec<String>,
```

Comment:
Why do we need multiple names? Inside of any gate creation, you can have multiple identities, but the name of the "constraints" can just be put once.
See:
```rust
fn configure(
        meta: &mut ConstraintSystem<F>,
        advice: [Column<Advice>; 2],
    ) -> <Self as Chip<F>>::Config {
        let s_add = meta.selector();

        // Define our addition gate!
        meta.create_gate("add", |meta| {
            let lhs = meta.query_advice(advice[0], Rotation::cur());
            let rhs = meta.query_advice(advice[1], Rotation::cur());
            let out = meta.query_advice(advice[0], Rotation::next());
            let s_add = meta.query_selector(s_add);

            vec![s_add * (lhs + rhs - out)]
        });

        // Define a second gate!
        meta.create_gate("second string", |meta| {
            let lhs = meta.query_advice(advice[0], Rotation::cur());
            let rhs = meta.query_advice(advice[1], Rotation::cur());
            let out = meta.query_advice(advice[0], Rotation::next());
            let s_add = meta.query_selector(s_add);

            vec![s_add * (lhs + rhs - out)]
        });

        AddConfig { advice, s_add }
    }
```

In this example of a `configure` call, you create two gates and each has only a single annotation. So why is the `Vec<String>` needed?

Reply:
The original `Gate` type had a vector of strings, so I kept it :P 

I found how this field is populated here: https://github.com/privacy-scaling-explorations/halo2/blob/73408a140737d8336490452193b21f5a7a94e7de/halo2_proofs/src/plonk/circuit.rs#L1942

I believe that in your example you're returning a vec of expresions, which implements `IntoIterator<Into<Constraint>>` because an expression has `Into<Constraint>` which sets the `name = ""`.

Reply:
LOL. Maybe we can ammend this in a sepparated PR. As it seems nonsense..

Reply:
Opened an issue to track this https://github.com/privacy-scaling-explorations/halo2/issues/249